File: 227-basic-calculator-ii/basic-calculator-ii.cpp
class Solution {
public:
    int calculate(string s) {
     int current = 0,previous =0 ,result =0;
     int i=0;
     char previous_op = '+';
     while(i<s.size()){
        char temp = s[i];
        if(isdigit(temp)){
            while(i<s.size() && isdigit(s[i])){
                  current =current*10 + (s[i]-'0');
                  i++;
            }
            i=i-1;// so that it doesn't go to next charector which is not number
            if(previous_op=='+'){
                result += current;
                previous = current;
            }
            else if(previous_op=='-'){
                result -=current;
                previous = -current;
            }
            else if(previous_op=='*'){
                result -= previous;
                result += previous*current;
                previous = previous*current;
            }
            else{
                result -= previous;
                result += previous/current;
                previous = previous/current;
            }
            current=0;
        }
        else if(s[i]!=' '){
        previous_op=s[i];
        }
        i++;
     }
     
        return result;
    }
};

File: 227-basic-calculator-ii/basic-calculator-ii.java
class Solution {
    public int calculate(String s) {
        char op = '+';
        int previous = 0;
        int current =0;
        int ans =0;
        int left = 0;
        while(left<s.length()){
            if(s.charAt(left)<='9' && s.charAt(left)>='0'){
               current =0;
               int i = left;
               while(i< s.length() && s.charAt(i)<='9' && s.charAt(i)>='0')
               {
                 int val = s.charAt(i)-'0';
                 current = current*10 + val;
                 i++;
               }
               left = i-1;
               if(op=='+'){
                ans +=current;
                previous = current;
               }
               else if(op=='-'){
                ans -=current;
                previous = -current;
               }
               else if(op=='*'){
                ans = ans - previous;
                ans += previous*current;
                previous = previous*current;
               }
               else {
                ans = ans - previous;
                ans += previous/current;
                previous = previous/current;
               }
            }
            else if(s.charAt(left)==' '){
            }
            else{
                op = s.charAt(left);
            }
            left++;
        }
        return ans;
        
    }
}